static/app/views/replays/utils/useReplayEvent.tsx

```diff
@@ -0,0 +1,180 @@
+import useApi from 'sentry/utils/useApi';
+
+export type RRWebEvents = ConstructorParameters<typeof RRWebPlayer>[0]['props']['events'];
+
+type State = {
+  /**
+   * The root replay event
+   */
+  event: undefined | Event;
+
+  /**
+   * If any request returned an error then nothing is being returned
+   */
+  fetchError: undefined | RequestError;
+
+  /**
+   * If a fetch is underway for the requested root reply.
+   * This includes fetched all the sub-resources like attachments and `sentry-replay-event`
+   */
+  fetching: boolean;
+
+  /**
+   * The list of related `sentry-replay-event` objects that were captured during this `sentry-replay`
+   */
+  replayEvents: undefined | Event[];
+
+  /**
+   * The flattened list of rrweb events. These are stored as multiple attachments on the root replay object: the `event` prop.
+   */
+  rrwebEvents: undefined | RRWebEvents;
+};
+
+type Options = {
+  /**
+   * When provided, fetches specified replay event by slug
+   */
+  eventSlug: string;
+
+  /**
+   *
+   */
+  location: any;
+
+  /**
+   *
+   */
+  orgId: string;
+};
+
+type Result = {
+  // TODO: non-state?
+} & Pick<State, 'fetchError' | 'fetching' | 'event' | 'rrwebEvents' | 'replayEvents'>;
+
+const IS_RRWEB_ATTACHMENT_FILENAME = /rrweb-[0-9]{13}.json/;
+function isRRWebEventAttachment(attachment: IssueAttachment) {
+  return IS_RRWEB_ATTACHMENT_FILENAME.test(attachment.name);
+}
+
+function useReplayEvent({eventSlug, location, orgId}: Options): Result {
+  const [projectId, eventId] = eventSlug.split(':');
+
+  const api = useApi();
+  const [state, setState] = useState<State>({
+    fetchError: undefined,
+    fetching: false,
+    event: undefined,
+    replayEvents: undefined,
+    rrwebEvents: undefined,
+  });
+
+  function fetchEvent() {
+    return api.requestPromise(
+      `/organizations/${orgId}/events/${eventSlug}/`
+    ) as Promise<Event>;
+  }
+
+  function fetchRRWebEvents() {
+    return api
+      .requestPromise(`/projects/${orgId}/${projectId}/events/${eventId}/attachments/`)
+      .then(attachments =>
+        Promise.all(
+          attachments.filter(isRRWebEventAttachment).map(attachment =>
+            api
+              .requestPromise(
+                `/api/0/projects/${orgId}/${projectId}/events/${eventId}/attachments/${attachment.id}/?download`
+              )
+              .then(resp => JSON.parse(resp))
+              .then(json => json.events)
+          )
+        )
+      )
+      .then(attachments => attachments.flat()) as Promise<RRWebEvents>;
+  }
+
+  function fetchReplayEvents() {
+    const replayEventsView = EventView.fromSavedQuery({
+      id: '',
+      name: '',
+      version: 2,
+      fields: ['timestamp', 'replayId'],
+      orderby: 'timestamp',
+      projects: [],
+      range: '14d',
+      query: `transaction:sentry-replay-event`,
+    });
+    replayEventsView.additionalConditions.addFilterValues('rootReplayId', [eventId]);
```

Comment:
Just a heads up this will be changing soon (rootReplayId and replayId will be the same)

Reply:
ack